build-tools/packages/build-infrastructure/src/fluidRepo.ts

```diff
@@ -235,3 +237,30 @@ export function getAllDependenciesInRepo(
 		workspaces: [...workspaces],
 	};
 }
+
+export async function setDependencyRange(
+	packagesToUpdate: IPackage[],
+	dependencies: IPackage[],
+	dependencyRange: InterdependencyRange
+): Promise<void> {
+	for (const pkg of packagesToUpdate) {
+		for (const dep of pkg.combinedDependencies) {
+			if (dependencies.some(d => d.name === dep.name)) {
```

Comment:
Consider making a Set out of the `dependencies` values, then use `.has`. It should be faster than `.some` on the array.